Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -440,12 +453,31 @@ class HTTP1ConnectionProvider {
 
     private func makeChannel(preference: HTTPClient.EventLoopPreference,
                              logger: Logger) -> EventLoopFuture<Channel> {
-        return NIOClientTCPBootstrap.makeHTTP1Channel(destination: self.key,
-                                                      eventLoop: self.eventLoop,
-                                                      configuration: self.configuration,
-                                                      sslContextCache: self.pool.sslContextCache,
-                                                      preference: preference,
-                                                      logger: logger)
+        let connectionID = HTTPConnectionPool.Connection.ID.globalGenerator.next()
+        let eventLoop = preference.bestEventLoop ?? self.eventLoop
+        let deadline = NIODeadline.now() + (self.configuration.timeout.connect ?? .seconds(10))
+        return self.factory.makeChannel(
+            connectionID: connectionID,
+            deadline: deadline,
+            eventLoop: eventLoop,
+            logger: logger
+        ).flatMapThrowing {
+            (channel, _) -> Channel in
+
+            // add the http1.1 channel handlers
+            let syncOperations = channel.pipeline.syncOperations
+            try syncOperations.addHTTPClientHandlers(leftOverBytesStrategy: .forwardBytes)
+
+            switch self.configuration.decompression {
+            case .disabled:
+                ()
+            case .enabled(let limit):
+                let decompressHandler = NIOHTTPResponseDecompressor(limit: limit)
+                try syncOperations.addHandler(decompressHandler)
+            }
+
+            return channel
```

Comment:
I'm a bit surprised to find this function here. Why is it being placed here, as opposed to being the responsibility of the factory object?

Reply:
Because this is functionality that is tied to the "current" implementation. This code will shall be removed as soon as we have the real deal lands. To keep the traffic in the factory low, I put it here. Happy to move though if you want me to.

Reply:
I think I'd like to keep it logically in the right place, even though I acknowledge it's only temporary.

Reply:
Fixed.